cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchConfig.java

```diff
@@ -35,37 +38,23 @@
     // a reverse proxy
     private URI rootUri;
 
-    // Optional custom headers
-    private Map<String, String> customHeaders;
+    private List<HttpConnectionRequestInterceptor> requestInterceptors = new ArrayList
+            <HttpConnectionRequestInterceptor>();
+    private List<HttpConnectionResponseInterceptor> responseInterceptors = new ArrayList
+            <HttpConnectionResponseInterceptor>();
 
-    private List<HttpConnectionRequestInterceptor> requestInterceptors;
-    private List<HttpConnectionResponseInterceptor> responseInterceptors;
+    private HttpConnectionRequestInterceptor headerInterceptor;
 
     public CouchConfig(URI rootUri) {
         this.rootUri = rootUri;
     }
 
-    public Map<String, String> getCustomHeaders() {
-        return customHeaders;
-    }
-
-    public void setCustomHeaders(Map<String, String> customHeaders) {
-        List<String> prohibitedHeaders = Arrays.asList("www-authenticate", "host", "connection",
-                "content-type", "accept", "content-length");
-        for (Map.Entry<String, String> header : customHeaders.entrySet()) {
-            if (prohibitedHeaders.contains(header.getKey().toLowerCase())) {
-                throw new IllegalArgumentException("Bad optional HTTP header: " + header);
-            }
-        }
-        this.customHeaders = customHeaders;
-    }
-
     public List<HttpConnectionRequestInterceptor> getRequestInterceptors() {
         return requestInterceptors;
     }
 
     public void setRequestInterceptors(List<HttpConnectionRequestInterceptor> requestInterceptors) {
-        this.requestInterceptors = requestInterceptors;
+        this.requestInterceptors.addAll(requestInterceptors);
```

Comment:
This change makes the method name controversial because it doesn't actually set the field any more. It should probably be renamed `addRequestInterceptors` (and similarly for `Response`).
This also means the interceptors cannot be replaced using the "setter". Is the expected pattern that if you want to change interceptors you manipulate the list from the `get*` method or that you create a new `CouchConfig`?